From 2fba3531204e31a18f4a4711295ec66273fda653 Mon Sep 17 00:00:00 2001
From: Matt Ellis <matell@microsoft.com>
Date: Mon, 22 May 2017 11:37:59 -0700
Subject: [PATCH] Don't pull down stage 0 from the internet

The composed build will set `PREBUILT_DOTNET_TOOL_DIR` to a location
where the dotnet tool can be copied from.

Long term we should get rid of the copy in favor of just using one
from a common location, but some of the Stage0 logic in the CLI build
system will need to change, I think.
---
 .../update-dependencies/update-dependencies.sh           | 11 ++++-------
 run-build.sh                                             | 16 +---------------
 2 files changed, 5 insertions(+), 22 deletions(-)

diff --git a/build_projects/update-dependencies/update-dependencies.sh b/build_projects/update-dependencies/update-dependencies.sh
index 4781c84..8a0d7f9 100755
--- a/build_projects/update-dependencies/update-dependencies.sh
+++ b/build_projects/update-dependencies/update-dependencies.sh
@@ -30,15 +30,12 @@ if [ -z "${DOTNET_INSTALL_DIR:-}" ]; then
    export DOTNET_INSTALL_DIR=$REPO_ROOT/.dotnet_stage0/x64
 fi
 
-# Install a stage 0
-echo "Installing .NET Core CLI Stage 0"
-$REPO_ROOT/scripts/obtain/dotnet-install.sh -Channel master -Architecture x64
-
-if [ $? -ne 0 ]; then
-    echo "Failed to install stage 0"
-    exit 1
+if [ ! -d "$DOTNET_INSTALL_DIR" ]; then
+    mkdir -p $DOTNET_INSTALL_DIR
 fi
 
+cp -r $DOTNET_TOOL_DIR/* $DOTNET_INSTALL_DIR/
+
 # Put the stage 0 on the path
 export PATH=$DOTNET_INSTALL_DIR:$PATH
 
diff --git a/run-build.sh b/run-build.sh
index 8f2f407..1a7bb9f 100755
--- a/run-build.sh
+++ b/run-build.sh
@@ -155,21 +155,7 @@ export VSTEST_TRACE_BUILD=1
 DOTNET_MULTILEVEL_LOOKUP=0
 
 # Install a stage 0
-(set -x ; "$REPOROOT/scripts/obtain/dotnet-install.sh" --channel "release/2.0.0" --install-dir "$DOTNET_INSTALL_DIR" --architecture "$ARCHITECTURE" $LINUX_PORTABLE_INSTALL_ARGS --version "2.0.0-preview2-006470")
-
-EXIT_CODE=$?
-if [ $EXIT_CODE != 0 ]; then
-    echo "run-build: Error: installing stage0 with exit code $EXIT_CODE." >&2
-    exit $EXIT_CODE
-fi
-
-# Install a project.json based CLI for use by tests
-(set -x ; "$REPOROOT/scripts/obtain/dotnet-install.sh" --channel "master" --install-dir "$DOTNET_INSTALL_DIR_PJ" --architecture "$ARCHITECTURE" --version "1.0.0-preview2-1-003177")
-EXIT_CODE=$?
-if [ $EXIT_CODE != 0 ]; then
-    echo "run-build: Error: installing project-json based cli failed with exit code $EXIT_CODE." >&2
-    exit $EXIT_CODE
-fi
+cp -r $DOTNET_TOOL_DIR/* $DOTNET_INSTALL_DIR/
 
 # Put stage 0 on the PATH (for this shell only)
 PATH="$DOTNET_INSTALL_DIR:$PATH"
-- 
1.8.3.1

